homeassistant/components/tankerkoenig/sensor.py

```diff
@@ -0,0 +1,100 @@
+"""Tankerkoenig sensor integration."""
+
+import logging
+
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    ATTR_LATITUDE,
+    ATTR_LONGITUDE,
+    STATE_CLOSED,
+    STATE_OPEN,
+    STATE_UNKNOWN,
+)
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, NAME
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_ADDRESS = "address"
+ATTR_BRAND = "brand"
+ATTR_FUEL_TYPE = "fuel_type"
+ATTR_IS_OPEN = "is_open"
+ATTR_STATION_NAME = "station_name"
+ATTRIBUTION = "Data provided by https://creativecommons.tankerkoenig.de"
+
+ICON = "mdi:fuel"
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the tankerkoenig sensors."""
+    tankerkoenig = hass.data[DOMAIN]
+    _LOGGER.debug("Setup platform. Stations: %s ", tankerkoenig.entity_list)
+    async_add_entities(tankerkoenig.entity_list)
+
+
+class FuelPriceSensor(Entity):
+    """Contains prices for fuel in a given station."""
+
+    def __init__(self, fuel_type, station, name=NAME):
+        """Initialize the sensor."""
+        self._station = station
+        self._station_id = station["id"]
+        self._fuel_type = fuel_type
+        self._name = name
+        self._latitude = station["lat"]
+        self._longitude = station["lng"]
+        self._is_open = STATE_OPEN if station["isOpen"] else STATE_CLOSED
+        self._address = f"{station['street']} {station['houseNumber']}, {station['postCode']} {station['place']}"
```

Comment:
Instead of concatenating a string of the address use a dictionary.

Reply:
What do you mean? Should I have one attribute for each part (i.e. one attribute "street", one attribute "houseNumber", ...)?

Reply:
No, I was thinking about something along the lines of:
```python
self._address = {
    "street": station["street"]
    "house_number":station["houseNumber"]
    "post_code": station["postCode"]
    "place": station["place"]
}
```

Reply:
But then I need to either concatenate the string anyway in the device_state_attributes method when filling the attributes, or create one attribute anyway for each address sub-field. Or is it possible to have a "dictionary" attribute? I don't remember seeing that on any of the integrations that I am using, but of course I haven't looked into all of them.

I'll create 4 different attributes for the different parts of the address, if it is ok with you.

Reply:
> I'll create 4 different attributes for the different parts of the address, if it is ok with you.

Yeah that is okay as well.